Compentio.SourceMapper.Tests/MapperGenerators/CodeSourceGeneratorTests.cs

```diff
@@ -0,0 +1,144 @@
+﻿using AutoFixture;
+using AutoFixture.AutoMoq;
+using Compentio.SourceMapper.Diagnostics;
+using Compentio.SourceMapper.Generators;
+using Compentio.SourceMapper.Metadata;
+using Compentio.SourceMapper.Processors.DependencyInjection;
+using FluentAssertions;
+using Microsoft.CodeAnalysis;
+using Moq;
+using System.Collections.Generic;
+using Xunit;
+
+namespace Compentio.SourceMapper.Tests.Generators
+{
+    public class CodeSourceGeneratorTests : CodeSourceGeneratorTestBase
+    {
+        private readonly IFixture _fixture;
+        private readonly Mock<IMapperMetadata> _mockMapperMetadata;
+        private readonly Mock<ISourcesMetadata> _mockSourcesMetadata;
+
+        private readonly CodeSourceGenerator _codeSourceGenerator;
+        private readonly GeneratorExecutionContext _generatorExecutionContext;
+
+        public CodeSourceGeneratorTests()
+        {
+            _fixture = new Fixture()
+                .Customize(new AutoMoqCustomization { ConfigureMembers = true })
+                .Customize(new SupportMutableValueTypesCustomization()); 
+
+            _mockMapperMetadata = _fixture.Create<Mock<IMapperMetadata>>();
+            _mockSourcesMetadata = _fixture.Create<Mock<ISourcesMetadata>>();
+            _mockSourcesMetadata.Setup(s => s.Mappers).Returns(new List<IMapperMetadata> { _mockMapperMetadata.Object });
+
+            _codeSourceGenerator = new CodeSourceGenerator(_mockSourcesMetadata.Object);
+            _generatorExecutionContext = GetFakeContext(GetTestCode());
```

Comment:
If we crete it in a base class, that it is also cen be placed in base class as protected